src/api/spec/models/project_spec.rb

```diff
@@ -408,20 +409,67 @@
   describe '#self.valid_name?' do
     context "invalid" do
       it{ expect(Project.valid_name?(10)).to be(false) }
-      it{ expect(Project.valid_name?('home:M0ses:raspi::qtdesktop')).to be(false) }
+      it "has ::" do
+        property_of {
+          s1 = sized(1){ string(/[a-zA-Z0-9]/) }
+          s2 = sized(range(1, 199)){ string(/[-+\w\.]/) }
+          s = s1 + s2
+          i = range(0, (s.length - 2))
```

Comment:
That is the index of the array where the :: will be inserted. If I do it with `s1 + s2 + ['::'] + s3` how do I generate `s2` and `s3`?